### PR TITLE
Tighten storage-operator permissions against Salt

### PR DESCRIPTION
**Pull request by gdemonet**
**Component**: salt, storage-operator

**Context**: see conversation in #2634

**Summary**: to do

**Acceptance criteria**: to do

---

Closes: #2634 
Fixes: #1528
Fixes: #2084

### CODE REVIEW

File: storage-operator/deploy/role.yaml [hunk 1]
```diff
@@ -4,36 +4,23 @@ metadata:
   creationTimestamp: null
   name: storage-operator
 rules:
+# For recording transition events
```
Comment (slaperche-scality) [inline storage-operator/deploy/role.yaml:7]: Good idea to add comments in this file :+1:

File: salt/metalk8s/salt/master/files/master-99-metalk8s.conf.j2 [hunk 2]
```diff
@@ -45,7 +45,9 @@ external_auth:
     storage-operator:
       - '*':
         - 'disk.dump'
-        - 'state.sls'
+        - 'state.sls':
```
Comment (slaperche-scality) [inline salt/metalk8s/salt/master/files/master-99-metalk8s.conf.j2:48]: Ho, this was simpler than I thought ^^

**Review by slaperche-scality — DISMISSED** [review]
LGTM

File: salt/metalk8s/salt/master/files/master-99-metalk8s.conf.j2 (hunk 2 above)
Comment (gdemonet) [reply salt/metalk8s/salt/master/files/master-99-metalk8s.conf.j2:48]: That's the theory, not working atm :sweat_smile:

Comment (gdemonet) [reply salt/metalk8s/salt/master/files/master-99-metalk8s.conf.j2:48]: OK it looks like it's working actually!

**Review by slaperche-scality — DISMISSED** [review]

File: storage-operator/deploy/role.yaml [hunk 3]
```diff
@@ -4,36 +4,63 @@ metadata:
...
 - apiGroups:
   - apps
   resources:
   - deployments
-  - daemonsets
   - replicasets
-  - statefulsets
   verbs:
-  - '*'
+  - get
```
Comment (NicolasT) [inline storage-operator/deploy/role.yaml:30]: Why does the storage-operator need to read deployments and replicasets?

File: storage-operator/deploy/role.yaml (hunk 1 above)
Comment (NicolasT) [reply storage-operator/deploy/role.yaml:7]: FWIW, if we update `operator-sdk`/`controller-runtime` (or move to `kubebuilder`), the RBAC rules are to be defined in the Go code using annotations, and this file is to be generated by the tooling.

File: storage-operator/deploy/role.yaml (hunk 3 above)
Comment (gdemonet) [reply storage-operator/deploy/role.yaml:30]: It comes from `kubemetrics`: when creating the Service, it reads its Pod's owner controller (a ReplicaSet) and this controller's controller (a Deployment) to determine the right `selector` to use, I think. The information is disseminated in several issues and pull requests in `operator-sdk`, and I found most of them when investigating the `Failed to list *unstructured.Unstructured` error logs. Could do some digging up my history if you want, but I can confirm that there were unauthorized errors if I remove those permissions.

**Review by slaperche-scality — APPROVED** [review]

File: storage-operator/deploy/role.yaml (hunk 1 above)
Comment (gdemonet) [reply storage-operator/deploy/role.yaml:7]: Oh yeah... Debt ticket #2644

**Pull request merged**